### PR TITLE
deprecate FairnessShuffler

### PR DESCRIPTION
**Pull request by msmouse**
## Description

Not gonna be used on testnet or mainnet any more.

## Type of Change
- [x] Refactoring

## Which Components or Systems Does This Change Impact?
- [x] Other (specify) -- tests and devnet genesis

## How Has This Been Tested?

build
compatibility tests

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: types/src/on_chain_config/execution_config.rs [hunk 1]
```diff
@@ -152,7 +148,7 @@ pub enum TransactionShufflerType {
     NoShuffling,
     DeprecatedSenderAwareV1(u32),
     SenderAwareV2(u32),
-    Fairness {
+    DeprecatedFairness {
```
Comment (zekun000) [inline types/src/on_chain_config/execution_config.rs:151]: why not just remove it if it's never enabled?

Comment (msmouse) [reply types/src/on_chain_config/execution_config.rs:151]: Thought we enabled it briefly on testnet?  @igor-aptos  do you remember?

Comment (igor-aptos) [reply types/src/on_chain_config/execution_config.rs:151]: yup enabled, found the perf issue, and reverted.

**Review by sitalkedia — APPROVED** [review]

**Review by igor-aptos — APPROVED** [review]
we need to keep the enum, but replay/state sync don't shuffle - so we don't need to keep the code - we can just remove it?

**Comment by msmouse** [comment]
@igor-aptos Compat won't pass because we've been using the FairnessShuffler as the genesis default.
Will land as is and remove code in a later release.

**Pull request merged**